### ISSUE TITLE
Issue when compiling Singularity

### ISSUE THREAD

**Issue by hempelc** [apptainer/singularity#5784]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
  3.7.0
```

### Expected behavior

Successfully compile singularity

### Actual behavior
```
$ make -C builddir
```
Results in
```
make: *** [Makefile:172: singularity] Error 1
make: Leaving directory '/hdd1/chempel/programs/singularity/builddir'
```

Detailed output:
```
$ cd singularity && \
./mconfig && make -C builddir && \
sudo make -C builddir install

Configuring for project `singularity' with languages: C, Golang
=> running pre-basechecks project specific checks ...
=> running base system checks ...
 checking: host C compiler... cc
 checking: host C++ compiler... c++
 checking: host Go compiler (at least version 1.13)... /usr/local/go/bin/go
 checking: host C compiler option -Wall... yes
 checking: host C compiler option -Werror... yes
 checking: host C compiler option -Wfatal-errors... yes
 checking: host C compiler option -Wno-unknown-warning-option... yes
 checking: host C compiler option -Wstrict-prototypes... yes
 checking: host C compiler option -Wpointer-arith... yes
 checking: host C compiler option -Wbad-function-cast... yes
 checking: host C compiler option -Woverlength-strings... yes
 checking: host C compiler option -Wframe-larger-than=2047... yes
 checking: host C compiler option -Wno-sign-compare... yes
 checking: host C compiler option -Wclobbered... yes
 checking: host C compiler option -Wempty-body... yes
 checking: host C compiler option -Wmissing-parameter-type... yes
 checking: host C compiler option -Wtype-limits... yes
 checking: host C compiler option -Wunused-parameter... yes
 checking: host C compiler option -Wunused-but-set-parameter... yes
 checking: host C compiler option -Wno-discarded-qualifiers... yes
 checking: host C compiler option -Wno-incompatible-pointer-types... yes
 checking: host C compiler option -pipe... yes
 checking: host C compiler option -fmessage-length=0... yes
 checking: host C compiler option -fPIC... yes
 checking: host `ar' path... ar
 checking: host `ld' path... ld
 checking: host `ranlib' path... ranlib
 checking: host `objcopy' path... objcopy
 checking: target C compiler... cc
 checking: target C++ compiler... c++
 checking: target `ar' path... ar
 checking: target `ld' path... ld
 checking: target `ranlib' path... ranlib
 checking: target `objcopy' path... objcopy
 checking: host compiles static binaries... yes
 checking: target compiles static binaries... yes
 checking: host os type... unix
 checking: host architecture... x86_64
 checking: target architecture... x86_64
 checking: host architecture word size... 64
 checking: target architecture word size... 64
 checking: project version... 3.7.0
 checking: project short version... 3.7.0
=> running post-basechecks project specific checks ...
 checking: namespace: CLONE_NEWPID... yes
 checking: namespace: CLONE_FS... yes
 checking: namespace: CLONE_NEWNS... yes
 checking: namespace: CLONE_NEWUSER... yes
 checking: namespace: CLONE_NEWIPC... yes
 checking: namespace: CLONE_NEWNET... yes
 checking: namespace: CLONE_NEWUTS... yes
 checking: namespace: CLONE_NEWCGROUP... yes
 checking: feature: NO_NEW_PRIVS... yes
 checking: feature: MS_SLAVE... yes
 checking: feature: MS_REC... yes
 checking: feature: MS_PRIVATE... yes
 checking: user capabilities... yes
 checking: header linux/securebits.h... yes
 checking: header linux/capability.h... yes
 checking: libseccomp+headers... yes
 checking: cryptsetup... yes
=> generating fragments ...
=> building Makefile ...
=> generating singularity.spec ...
=> project singularity setup with :
    - host arch: x86_64
    - host wordsize: 64-bit
    - host C compiler: cc
    - host Go compiler: /usr/local/go/bin/go
    - host system: unix
      ---
    - target arch: x86_64
    - target wordsize: 64-bit
    - target C compiler: cc
      ---
    - config profile: release
      ---
    - SUID install: yes
    - Network plugins: yes
      ---
    - verbose: no
      ---
    - cryptsetup: /sbin/cryptsetup
      ---
    - version: 3.7.0
=> /hdd1/chempel/programs/singularity/builddir/Makefile ready, try:
   $ cd /hdd1/chempel/programs/singularity/builddir
   $ make
make: Entering directory '/hdd1/chempel/programs/singularity/builddir'
 GEN GO DEP /hdd1/chempel/programs/singularity/builddir/starter.d
 GEN GO DEP /hdd1/chempel/programs/singularity/builddir/singularity.d
 GO singularity
    [+] GO_TAGS "containers_image_openpgp sylog oci_engine singularity_engine fakeroot_engine apparmor selinux seccomp"
../internal/pkg/build/sources/oci_unpack_linux.go:21:2: cannot find package "." in:
	/hdd1/chempel/programs/singularity/vendor/github.com/openSUSE/umoci
../internal/pkg/build/sources/oci_unpack_linux.go:22:2: cannot find package "." in:
	/hdd1/chempel/programs/singularity/vendor/github.com/openSUSE/umoci/oci/layer
../internal/pkg/build/sources/oci_unpack_linux.go:23:2: cannot find package "." in:
	/hdd1/chempel/programs/singularity/vendor/github.com/openSUSE/umoci/pkg/idtools
make: *** [Makefile:172: singularity] Error 1
make: Leaving directory '/hdd1/chempel/programs/singularity/builddir'
```

### Steps to reproduce this behavior

Following steps on the [website](https://sylabs.io/guides/3.7/user-guide/quick_start.html#quick-installation-steps):
```
$ sudo apt-get update && sudo apt-get install -y \
  build-essential \
  libssl-dev \
  uuid-dev \
  libgpgme11-dev \
  squashfs-tools \
  libseccomp-dev \
  wget \
  pkg-config \
  git \
  cryptsetup

$ wget https://golang.org/dl/go1.15.6.linux-amd64.tar.gz && \
sudo tar -C /usr/local -zxf go1.15.6.linux-amd64.tar.gz
$ echo 'export PATH=/usr/local/go/bin:$PATH' >> ~/.bashrc && \
source ~/.bashrc
$ wget https://github.com/hpcng/singularity/releases/download/v3.7.0/singularity-3.7.0.tar.gz && \
tar -zxf singularity-3.7.0.tar.gz
$ cd singularity && \
./mconfig && make -C builddir && \
sudo make -C builddir install
```

Note: I used this exact code on a fresh Amazon Web Server using the Amazon Machine Image Ubuntu 20.04 LTS 64-bit x86 and it runs without problems.

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

```

### How did you install Singularity

See above.

Thanks!

**Comment by dtrudg** [comment]
Do you have a symlink in this path, or in your GOPATH? The Go tools do not follow symlinks as might be expected, and this is a relatively common cause of problems when building.

**Comment by hempelc** [comment]
Hey David, thanks for your quick reply! I already installed and uninstalled Singularity together with go a while ago. Back then, I was manually setting the GOPATH and it was pointing to a not-anymore-existing directory. Changed that and now it works! Thanks!

**Comment by Statekeylablin** [comment]
Hi hempelc, I‘ve met the same problem. Could you please tell me how to solve it?

**Comment by hempelc** [comment]
Hey Statekeylablin, the problem occurred because I had installed "go" a while ago. go uses a GOPATH in the user profile, and back then, my GOPATH was pointing to a directory that didn't exist anymore. So to fix the issue, I uninstalled go and manually deleted the GOPATH out of my profile (~/.bashrc or ~/.bash_profile). Then I followed the installation instructions for Singularity and it worked! I hope that helps.
